src/query/remote/compressed_codecs.go

```diff
@@ -169,13 +172,15 @@ func compressedSeriesFromSeriesIterator(
 			if err != nil {
 				return nil, err
 			}
-
 			replicaSegments = append(replicaSegments, segments)
 		}
 
-		compressedReplicas = append(compressedReplicas, &rpc.M3CompressedValuesReplica{
+		readers.Rewind()
```

Comment:
Can we add a test that would ensure that the series iterators being fed in are still iteratable and readable after compression?